### ISSUE TITLE
distance metrics should be usable as sklearn.neighbors.DistanceMetric objects

### ISSUE THREAD

**Issue by gregcaporaso** [biocore/scikit-bio#1042]
We should ultimately support passing our distance metrics (e.g., unifrac) as [sklearn.neighbors.DistanceMetric](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.DistanceMetric.html) object, for use with (e.g.) nearest neighbor clustering. @justin212k is doing some work with these sklearn methods.

**Comment by jairideout** [comment]
I think this is already supported by scikit-learn and shouldn't require any changes on our end:

```python
from sklearn.neighbors import DistanceMetric
dist = DistanceMetric.get_metric('pyfunc', func=lambda a, b: 42) # pass UniFrac function here
```

See the two answers in [this SO post](http://stackoverflow.com/questions/21052509/sklearn-knn-usage-with-a-user-defined-metric) for examples of using nearest neighbor clustering with custom distance metrics, including how to handle metrics that accept additional arguments (e.g., a tree).

@gregcaporaso can this be closed?

**Comment by gregcaporaso** [comment]
That makes sense, thanks.
